### ISSUE TITLE
[Feature Request] Make `RocketConfig::parse` public

### ISSUE THREAD

**Issue by lawliet89** [SergioBenitez/Rocket#228]
... or something similar for the `rocket::config::Config` struct like:

 - Deriving/implementing `Deserialize` for `rocket::config::Config` 
 - Moving the parsing method to `rocket::config::Config` and making it public

## Why you believe this feature is necessary and use case

I am trying to implement a Rocket application with additional configuration options. Currently, the `Config` struct only supports a simple key-value store for extra configuration options. That is fine.

As an alternative, I am implementing a separate configuration structure that will be deserialized from another JSON/TOML file. 

I would like to allow users to specify the filename for this additional `JSON/TOML` file as one of the extra configuration options in `Rocket.toml`. This file can be, and should probably be, different for the three rocket environments.

Then, I would like to parse `Rocket.toml` with the same rules that Rocket does, and extract this extra value BEFORE I `launch` Rocket so that I can deserialize the JSON/TOML file and then add the struct into Rocket as to be `manage`d. 

This is not really do-able at the moment, or even to implement for myself because  `rocket::config::Config`  does not derive `Deserialize`. Alternative, I would have to essentially duplicate quite a bit of code to achieve this.

**Comment by SergioBenitez** [comment]
I'm not sure which form this will come in, but I believe something like this should exist.

Slating for 0.4.

**Comment by LukasKalbertodt** [comment]
To give another use case example:

In my **build script**, I compile SCSS files to CSS files. Depending on the Cargo profile (debug/release) I choose whether or not to minify the resulting CSS. It would be awesome to depend on the Rocket environment instead (or additionally). Something like "minify CSS in production". I imagine something like this:

```rust
// Creates the config just as rocket would normally do: `Rocket.toml`, env vars, ...
let rocket_config = rocket::Config::from_env();

if env::var("PROFILE").unwrap() == "release" && rocket_config.environment.is_prod() {
    // minify CSS
}
```

**Comment by new-commer** [comment]
This issue seems has not been resolved in 0,4 , and i think this  issue  is necessary, just change  `parse` to public . We looking forward  it。

**Comment by SergioBenitez** [comment]
Let's fix this in #852.
